.github/workflows/build.yaml

```diff
@@ -44,5 +59,40 @@ jobs:
         run: cartesi doctor
 
       - name: Build
-        run: cartesi build
-        working-directory: ${{ matrix.template }}
+        run: |
+          cartesi build
+          cartesi hash
+        working-directory: ${{ matrix.allTemplates }}
+
+  build-on-pr:
```

Comment:
Instead of having different jobs for main and PR why don't you just manipulate the matrix based on that, and use the same job?

Reply:
I create two variables to be used as matrix, I could think of how to use both in a single job.